src/librustc_typeck/check/closure.rs

```diff
@@ -266,7 +266,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         let trait_ref = projection.to_poly_trait_ref(tcx);
 
         let is_fn = tcx.lang_items().fn_trait_kind(trait_ref.def_id()).is_some();
-        let gen_trait = tcx.lang_items().gen_trait().unwrap();
+        let gen_trait = tcx.require_lang_item(crate::middle::lang_items::LangItem::GeneratorTraitLangItem);
```

Comment:
```suggestion
        let gen_trait = tcx.require_lang_item(lang_items::GeneratorTraitLangItem);
```
(convert this to an import)